src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -0,0 +1,88 @@
+package com.hibob.academy.dao
+
+import org.jooq.Record
+import org.jooq.DSLContext
+import org.jooq.RecordMapper
+
+
+class PetDao(private val sql: DSLContext) {
+    private val pet = PetTable.instance
+    private val petDataMapper= RecordMapper<Record, PetData> { record ->
+        PetData(
+            record[pet.petName],
+            record[pet.dateOfArrival],
+            record[pet.companyId],
+        )
+    }
+
+    private val petAllInfoMapper = RecordMapper<Record, PetAllInfo>{ record ->
+        PetAllInfo(
+            record[pet.id],
+            record[pet.petName],
+            record[pet.dateOfArrival],
+            record[pet.companyId],
+            PetType.valueOf(record[pet.petType].uppercase()),  // Convert string to enum (assuming your DB stores it as a string)
+            record[pet.ownerId]
+        )
+    }
```

Comment:
Why do we need 2 mappers and 2 data class?
Why not use only 1 mapper?

Reply:
Got it!